### PR TITLE
Yugabyte DB operator implementation

### PR DESCRIPTION
**Pull request by samkulkarni20**
Enabling TLS communication across cluster currently fails. Though the code review can start while I keep working on fixing it.

### CODE REVIEW

File: deploy/crds/yugabyte_v1alpha1_ybcluster_cr.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: yugabyte.com/v1alpha1
+kind: YBCluster
+metadata:
+  name: example-ybcluster
```
Comment (ramkumarvs) [inline deploy/crds/yugabyte_v1alpha1_ybcluster_cr.yaml:4]: not sure if we want to call this example-ybcluster. YB universe is the construct we use from within YugaWare. Would be good to run it by @rkarthik007 and see how we want to name this?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+require (
+	github.com/Masterminds/sprig v0.0.0-20190301161902-9f8fceff796f
+	github.com/NYTimes/gziphandler v1.0.1 // indirect
+	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
+	github.com/go-ini/ini v1.46.0 // indirect
+	github.com/go-logr/logr v0.1.0
+	github.com/go-openapi/spec v0.19.0
+	github.com/masterminds/sprig v2.20.0+incompatible
+	github.com/operator-framework/operator-sdk v0.10.1-0.20190912205659-c084b570a6af
+	github.com/rook/rook v1.1.0
```
Comment (ramkumarvs) [inline go.mod:12]: why are we using rook here?

File: README.md [hunk 3]
```diff
@@ -1 +1,93 @@
 # Yugabyte Kubernetes Operator
+
```
Comment (ramkumarvs) [inline README.md:2]: As synced up offline, we need to capture more information around pre-req on operator-sdk and also instructions on how to build and deploy and replace the image_tag etc needs to be documented.

**Review by ramkumarvs — COMMENTED** [review]
Overall looks good, we can follow up and add the TLS, Just address some comments and we can land this.

File: README.md (hunk 3 above)
Comment (samkulkarni20) [reply README.md:2]: Added `CONTRIBUTING.md` to the PR

File: go.mod (hunk 2 above)
Comment (samkulkarni20) [reply go.mod:12]: It must have been imported by the framework. I'm not using anything from rook codebase for operator logic.

**Review by ramkumarvs — APPROVED** [review]

**Pull request merged**